Konvoy2.1 - D2IQ-82072 Azure Installer: Add missing docs information (#3979)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.1/choose-infrastructure/azure/advanced/new/index.md

    CLUSTER_NAME=my-azure-cluster
    ```

### Naming Tips and Tricks

1.  To create a cluster name that is unique, use the following command:

    ```sh
    CLUSTER_NAME=$(whoami)-azure-cluster-$(LC_CTYPE=C tr -dc 'a-z0-9' </dev/urandom | fold -w 5 | head -n1)
    echo $CLUSTER_NAME
    ```

    The output appears similar to:

    ```text
    hunter-azure-cluster-pf4a3
    ```

    This command creates a unique name every time you run it, so use it carefully.

1.  Set the environment variable to the name you assigned this cluster:

    ```sh
    CLUSTER_NAME=my-azure-cluster
    ```

## Create new Azure Kubernetes cluster objects

> This procedure uses the `--dry-run` and `--output-yaml` flags together to create basic Azure Kubernetes cluster objects in a YAML file. This approach allows you to examine the YAML objects before creating the actual Azure Kubernetes cluster itself. For example, you may need to specify an HTTP proxy for your control plane and worker nodes. You could also configure the cluster to use your existing network infrastructure. If you need either of those additional configurations, skip to the relevant procedures _without_ executing this one.

Reviewer: The last sentence here is incorrect because you should only skip this step if you need an http proxy.   The "Existing network" section does not instruct you to create the objects, so if you skip from here to that section, you won't have a configuration to inspect and edit.   Also, I find the 'choose your own adventure' format here to be kind of verbose and confusing.   There's really just two different starting points here (either with or without a proxy) and then an optional step where you can mess with the network config if you need to use existing network infra)...

```suggestion
This procedure uses the `--dry-run` and `--output-yaml` flags together to create basic Azure Kubernetes cluster objects in a YAML file. This approach allows you to examine the YAML objects before creating the actual Azure Kubernetes cluster itself.   

When creating the basic Azure Kubernetes cluster objects, you need to first consider if you need to use an HTTP proxy.   If you do so, then you will need to do some additional configuration when creating the cluster objects.   Consult the "Configure the control plane and workers to use an HTTP Proxy" section for more details. 
```

Author: Resolved as noted in my local branch.

Author: Ideally, we would break up these several headings into several different topics and cross-link between them as needed to get the right selections into the cluster. I am loathe to nest too many levels down in the navigation hierarchy, lest headings get lost from display and search. They are still somewhat "lost" here, too, but at least they are in one place, and we have the internal links for this page. Not the best solution, but enough perhaps with the release just 2 days off.

    > ${CLUSTER_NAME}.yaml
    ```

> The output of this command is `${CLUSTER_NAME}.yaml` file that you can examine or modify. If you use this method to create a basic cluster without HTTP proxies or setting existing network infrastructure values, skip to the heading, "Inspect or edit the cluster objects."

Reviewer: ```suggestion
The output of this command is `${CLUSTER_NAME}.yaml` file that you can examine or modify.  If you use this method to create a basic cluster without HTTP proxies, skip to the heading, "Inspect or edit the cluster objects."
```

Author: Resolved as noted in my local branch.

## (Optional) Configure the control plane and workers to use an HTTP Proxy

> This procedure uses the `--dry-run` and `--output-yaml` flags together to create basic Azure Kubernetes cluster objects in a YAML file. This approach allows you to examine the YAML objects before creating the actual Azure Kubernetes cluster itself. To configure the Control Plane and Worker nodes to use an HTTP proxy:

Reviewer: This is a bit of 'department of redundancy department', we just said most of this in the last section:

```suggestion
To configure the Control Plane and Worker nodes to use an HTTP proxy:
```

Author: Resolved as noted in my local branch.

    > ${CLUSTER_NAME}.yaml
    ```

> The output of this command is `${CLUSTER_NAME}.yaml` file that you can examine or modify further. If you are using this method to create your cluster without setting existing network infrastructure values, review the information under "Inspect or edit the cluster objects" and then go to "Create the actual Kubernetes cluster."

Reviewer: I don't think the last sentence adds much info here, I'd suggest instead of adding that extra info here, we just add something to the "configure existing network infrastructure" below that says "if you don't need this, you can skip it".  See below

```suggestion
The output of this command is `${CLUSTER_NAME}.yaml` file that you can examine or modify further. 
```

Author: Resolved as noted in my local branch.

    1.  Node Pool

        A Node Pool is a collection of machines with identical properties. For example, a cluster might have one Node Pool with large memory capacity, another Node Pool with GPU support. Each Node Pool is described by three objects: The MachinePool references an object that describes the configuration of Kubernetes components (for example, the kubelet) deployed on each node pool machine, and an infrastructure-specific object that describes the properties of all node pool machines. Here, it references a _KubeadmConfigTemplate_, and an _AzureMachineTemplate_ object, which describes the instance type, the type of disk used, the size of the disk, among other properties.

    For in-depth documentation about the objects, read [Concepts][capi_concepts] in the Cluster API Book.

> ## Configure existing network infrastructure in the cluster

Reviewer: ```suggestion
## (Optional) Configure existing network infrastructure in the cluster
```

> As part of inspecting and editing your cluster objects, you can also configure it to use existing network infrastructure.

Reviewer: ```suggestion
As part of inspecting and editing your cluster objects, you can also configure it to use existing network infrastructure.   If you do not need to use existing network infrastructure, you can skip this step.
```

Author: Resolved as noted in my local branch.
